File: packages/java/tests/spring/react-grid-test/src/main/java/com/vaadin/hilla/test/reactgrid/AbstractEntity.java
package com.vaadin.hilla.test.reactgrid;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;
import jakarta.persistence.Version;

import com.vaadin.hilla.Nullable;

@MappedSuperclass
public class AbstractEntity {
    @Id
    @Nullable
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Version
    @Nullable
    private Long version;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getVersion() {
        return version;
    }

    public void setVersion(Long version) {
        this.version = version;
    }
}

File: packages/java/tests/gradle/kotlin-gradle-test/src/main/kotlin/com/vaadin/hilla/kotlin/nullable/test/data/AbstractEntity.kt
package com.vaadin.hilla.kotlin.nullable.test.data

import jakarta.persistence.*
import java.util.concurrent.atomic.AtomicReference

@MappedSuperclass
abstract class AbstractEntity<ID>(
    @Version
    val version: Int = 0,
) {
    @Id
    @GeneratedValue(
        strategy = GenerationType.SEQUENCE,
        generator = "idgenerator"
    )
    @SequenceGenerator(name = "idgenerator", initialValue = 1000)
    var id: AtomicReference<ID> = AtomicReference()

    override fun hashCode(): Int = id?.hashCode() ?: 0

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || other !is AbstractEntity<*>) return false
        return id == other.id
    }
}
